ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,810 @@
+    my $packages_modified = 0;
+
+    if ( !defined($status) ) {
+	my @lines = split /\n/, $yum_error;
+	foreach my $line (@lines) {
+	    my $cf = index($line, 'conflicts with file from package');
+	    if ( $cf != -1) {
+		my $pkg = $line;
+		$pkg =~ s,^.*package ,,;
+		$pkg =~ s,-[0-9]*:,-,;
+		$self->execute_yum_command(["rpm -e --nodeps $pkg"], "removing $pkg to avoid file conflict");
+		$packages_modified = 1;
+	    }
+	}
+	if ( $packages_modified ) {
+	    $status = $self->execute_yum_command($command, 'retrying: '.$why, $keeps_state, $stdin, $error_logger, $error_ok);
+	}
+    }
+    return $status;
+}
+
+sub Configure
+{
+    my ( $self, $config ) = @_;
+
+    # Make sure stdout and stderr is flushed every time to not to
+    # put mess on serial console
+    autoflush STDOUT 1;
+    autoflush STDERR 1;
+
+    # We are parsing some outputs in this component. We must set a
+    # locale that we can understand.
+    local $ENV{LANG}   = 'C';
+    local $ENV{LC_ALL} = 'C';
+
+    my $repos = $config->getElement(REPOS_TREE)->getTree();
+    my $t     = $config->getElement(CMP_TREE)->getTree();
+    
+    # Display system info
+    if ( defined($t->{quattor_os_release}) ) {
+	$self->info("target OS build: ", $t->{quattor_os_release});
+    }
+    
+    my $spmaver = $self->execute_yum_command( ["rpm -q --qf %{NAME}-%{VERSION}-%{RELEASE}.%{ARCH} ncm-spma"], "checking for spma version", 1);
+    $self->info("SPMA version: ", $spmaver);
+    $self->info("User packages permitted: $t->{userpkgs}");
+
+    # Convert these crappily-defined fields into real Perl booleans.
+    $t->{run}      = $t->{run} eq 'yes';
+    $t->{userpkgs} = defined( $t->{userpkgs} ) && $t->{userpkgs} eq 'yes';
+
+    # test for sharp spma execution early
+    my $spma_run = 0;
+    if ( -e "/.spma-run" ) {
+        if ( !unlink "/.spma-run" ) {
+            $self->warn("unable to remove file /.spma-run: $!");
+        }
+	$spma_run = 1;
+    }
+
+    # Detect legacy OS (RHEL5 and older)
+    my $legacy_os = 0;
+    my $fhi;
+    if ( open( $fhi, '<', "/etc/redhat-release" ) ) {
+        while ( my $line = <$fhi> ) {
+            if ( index( $line, 'release 5' ) >= 0 ) {
+                $legacy_os = 1;
+                last;
+            }
+        }
+        $fhi->close();
+    } else {
+        $self->warn("Unable to determine OS release.");
+    }
+
+    # Generate YUM config file
+    my $yum_conf_file = CAF::FileWriter->new( YUM_CONF_FILE, log => $self );
+    my $excludes      = $t->{excludes};
+    print $yum_conf_file $t->{yumconf};
+    print $yum_conf_file "exclude=" . join ( " ", sort @$excludes );
+    $yum_conf_file->close();
+
+    if ( !$NoAction ) {
+        # Remove all repositories if userpkgs is not defined
+        $self->cleanup_old_repos( REPOS_DIR, $repos, $t->{userpkgs} ) or return 0;
+    }
+
+    # Generate new installation repositories from host profile
+    my $proxy = !defined($t->{proxy}) ? 'yes' : $t->{proxy};
```

Comment:
I followed the old userpkgs check from the old code. No problem to simplify that.